### PR TITLE
Add status and type filter fields to vulnerability endpoint

### PR DESCRIPTION
**Pull request by Espectro123**
|Related issue|
|---|
| #7972 |

## Description

Hello team!

In this issue I have added two parameters for the endpoint ```/vulnerability/{agent_id}```

Both parameters are part of the filters available to make calls to this endpoint.

The **Type** parameter will allow us to filter the vulnerabilities by their type: ```OS```, ```PACKAGE```

The **Status** parameter will allow us to filter the vulnerabilities by their status: ```VALID```, ```PENDING```, ```OBSOLETE```

#### framework/wazuh/core/cluster/tests/
```
=== 55 passed, 4 warnings in 0.39s ===
```
#### framework/wazuh/core/cluster/dapi/tests/
```
=== 23 passed, 9 warnings in 0.52s ===
```
#### framework/wazuh/core/tests/
```
=== 654 passed, 9 warnings in 2.16s ===
```
#### framework/wazuh/tests/
```
=== 764 passed, 11 warnings in 62.76s (0:01:02) ===
```
#### framework/wazuh/rbac/tests/

```
=== 227 passed, 3 warnings in 170.78s (0:02:50) ===
```
#### api/api/test/

```
=== 225 passed, 16 warnings in 1.06s ===
```

### test_vulnerability_endpoints
```
=== 1 passed, 3 warnings in 374.80s (0:06:14) ===
```

### test_rbac_white_vulnerability_endpoints
```
=== 1 passed, 3 warnings in 154.86s (0:02:34) ===
```
### test_rbac_black_vulnerability_endpoints
```
=== 1 passed, 3 warnings in 171.17s (0:02:51) ===
```

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [ ] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [ ] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

### CODE REVIEW

File: framework/wazuh/tests/data/schema_cve_test.sql [hunk 1]
```diff
@@ -11,6 +11,8 @@ CREATE TABLE IF NOT EXISTS vuln_cves (
     version TEXT,
     architecture TEXT,
     cve TEXT,
+    type TEXT,
```
Comment (davidjiglesias) [inline framework/wazuh/tests/data/schema_cve_test.sql:14]: Add new type and status values to schema below

File: framework/wazuh/tests/test_vulnerability.py [hunk 2]
```diff
@@ -58,6 +58,8 @@ def send_msg_to_wdb(msg, raw=False):
     ({'filters': {'version': '0.1.0', 'architecture': 'ARM'}}, 'cve', ['CVE-2019-1020019']),
     ({'filters': {'cve': 'CVE-2019-1020016'}}, 'cve', ['CVE-2019-1020016']),
     ({'filters': {'cve': 'CVE-2019-1020016', 'architecture': 'ARM'}}, 'cve', []),
+    ({'filters': {'status': 'OBSOLETE'}},'cve',[]),
```
Comment (davidjiglesias) [inline framework/wazuh/tests/test_vulnerability.py:61]: add more a couple more filters with actual results and not empty cve, based on previously requested changes in schema

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: framework/wazuh/tests/data/schema_cve_test.sql (hunk 1 above)
Comment (Espectro123) [reply framework/wazuh/tests/data/schema_cve_test.sql:14]: Arreglado

File: framework/wazuh/tests/test_vulnerability.py (hunk 2 above)
Comment (Espectro123) [reply framework/wazuh/tests/test_vulnerability.py]: Arreglado

**Review by davidjiglesias — APPROVED** [review]

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**